### PR TITLE
feat: add right to left support with css logical properties

### PR DESCRIPTION
**Pull request by nabaroa**
## 🖼 Context

RTL support.

### Beta releases

```
 - @transferwise/components@36.10.1-beta.49
 - @transferwise/neptune-css@9.2.3-beta.51
 - @transferwise/dynamic-flows@13.8.10-beta.47
```

### Tasks status:

1. Alignment utilities (text-xs-left...) with logical values :white_check_mark:
2. Spacing utilities (m-r-1...) with logical properties :white_check_mark:
3. Float utilities (pull-xs-left...) with logical values (moved from bootstrap to neptune-css) :white_check_mark:
4. Replaced alignment, spacing and positions (left:...) from all neptune-css and components :white_check_mark:
5. Removed rtl logic from react components and fixed the tests :white_check_mark:
6. grid column system :white_check_mark:
7. Navbar with logical properties/values :white_check_mark:
8. Dropdown with logical properties/values :white_check_mark:
9. Borders :white_check_mark:
10. Update margin/padding shorthands when right/left values were different :white_check_mark: 
11. Add a mixing to control the horizontal directions properties and the support :white_check_mark:
12. Support to flow-relative values  :white_check_mark:
13. Support to inset-inline (left:, right;) properties for IE11 :white_check_mark:
14. Beta test  :white_check_mark:
15. Documentation  :white_check_mark:
16. Tooltip and popover have absolute positions inlined. Need to sort it out with JS?  :white_check_mark:
17. Stepper [source line]. Sort out the step positions in RTL.  :white_check_mark:
18. ~Add default dir="ltr" attribute to Storybook.~ Changed addon in Storybook to allow dir attribute in html ([recommended by W3C](https://www.w3.org/International/questions/qa-html-dir#quickanswer)) tag and not in body. :white_check_mark: 
19. Fix stylelint. ✅ 
20. Visual tests. :joe-sayer:   :white_check_mark:
21. ~Allow default state without dir attribute in html tag for those consumers they don’t need RTL approach by now.~. Discarded option because[ it’s is expensive KB wise](https://docs.google.com/spreadsheets/d/1epMGFnp2wlSa_H3lSPCbms_Z68QgqoavxhVJFY5ZZOY/edit#gid=0)  :white_check_mark: 
22. Neptune.css bundle is 81KBs (7KB gzipped) heavier. Any strategy to avoid this? Split the rtl supported bundle? -> We can go with the **RTL Support for IE option**, which is 53KSs heavier for all cases. IE support is going to be removed eventually so the extra KBs it’s a temporary fact. We have to create a simple consume option. :white_check_mark: 
23. Import css components from bootstrap. At least the ones which have physical horizontal properties in there. ✅ 

----

### Requirements:

- The consumer must add `dir=”ltr”` or `dir=”rtl”` in HTML tag depending the language.
- New syntax to write CSS for alignment, float, clear, margin | padding in Neptune-web, to get bidirectional options.
    E.g.
```css
   .margin(right, var(--size-8));
   .float(left);
   .text-align(right);
   .left(100px);
```

- The mixin [_logical-properties.less](https://github.com/transferwise/neptune-web/blob/e4199c0fdfcd0f247bb774681c0f10a73ef69f5f/packages/css/src/less/mixins/_logical-properties.less) for rtl support gives us control to generate the output we want for directional declarations. 
- The neptune.css bundle will have +40KB. We could leave it lighter (~+20KB) if we decide to remove rtl support for IE11 as a graceful degratation.
- [RTL KB increase table](https://docs.google.com/spreadsheets/d/1epMGFnp2wlSa_H3lSPCbms_Z68QgqoavxhVJFY5ZZOY/edit#gid=0)

## New rules to create CSS for RTL:

There is a mixin to control the direction of the physical horizontal properties and values in Neptune. With this mixin we are going to be able to control and scale the support in browsers of logical properties. Right now it’s prepared to output the IE11 friendly version.

### Usage

Add the import reference at the top of the .less file:

```
@import (reference)
  "@transferwise/neptune-css/src/less/mixins/_logical-properties.less";
```

Update all the horizontal properties with a mixin call in your custom .less file:
Tip: If left and right values are the same we can keep them as they are without having to update to the mixin syntax. E.g: 

```
padding-left: var(--foo1);
padding-right: var(--foo1);
```

#### Alignment:

`text-align: left; → .text-align(left);`

#### Spacing:

```
margin-left: var(--foo); → .margin(left, var(--foo));

padding-left: var(--foo); → .padding(left, var(--foo));
```

#### Positions: 

`left: var(foo);. → .left(var(--foo)); `

#### Floats:

`float: left;. → .float(left);`

#### Clear:

`clear: left → .clear(left);`

#### Border: 

```
border-left: 1px solid pink; → .border(left, 1px, solid, pink);

border-left-width: 1px; → .border-width(left, 1px);

border-bottom-left-radius: 3px; → .border-radius(left, 3px);
```

### Collateral changes:

- Moved some components from `Bootstrap` to `Neptune-css`:
    close.less
    navs.less
    column-layout.less
    list-group.less
    dropdowns.less
    circles.less
    panels.less
    droppable.less
    modals.less
    navbar.less (renamed to navbar-base.less)
    utilities.less
    popovers.less
    process.less
    sequences.less
    utilities.less
    tooltip.less
    mixin/_circle.less
    mixin/_list-group.less
    mixin/panels.less
    mixin/reset-text.less
    mixin/sequence.less

Didn’t translate all the variables in `navbar-base.less` and `dropdowns.less`. It’s better to split that task into [this ticket](https://transferwise.atlassian.net/browse/DS-795).
- Removed vendor prefixes from modal, navbar
- Added .stylintignore to avoid inconsistent linter issues in navbar-base.less, which were avoiding the commit.
- Removed .caret styles from dropdowns.less
- Removed deprecated:
    -.droppable-active-cover. Deprecated 5 years ago. 0 results in code https://codesearch.tw.ee/. Only CSS references.
    -.glyphicon
    -opacity mixin. Not needed filter for IE8. Modal and tooltip
-------

## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/.storybook/storybook.css [hunk 1]
```diff
@@ -1,3 +1,7 @@
 .simplebar-content pre {
   background-color: #333;
 }
+
+[dir="rtl"] {
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (nabaroa) [reply packages/components/.storybook/storybook.css]: Removed. Unnecessary. Fixed [in this commit](https://github.com/transferwise/neptune-web/pull/1053/commits/a3388241e21db7f304e915e95d57a16552ddb812).

File: packages/components/src/switch/Switch.less [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';
+@import (reference) '@transferwise/neptune-css/src/less/mixins/_logical-properties.less';
```
[the remark this answers is not in the document]
Comment (nabaroa) [reply packages/components/src/switch/Switch.less:2]: Fixed in [this commit.](https://github.com/transferwise/neptune-web/pull/1053/commits/2551ca3f7bb3ff03121fdaf00066f05d33a15758)

File: packages/components/src/checkboxButton/CheckboxButton.less [hunk 3]
```diff
@@ -21,7 +22,8 @@
 .checkbox .np-checkbox-button {
   input[type='checkbox'] {
     top: 0;
-    left: 0;
+    .left(0);
```
[the remark this answers is not in the document]
Comment (nabaroa) [reply packages/components/src/checkboxButton/CheckboxButton.less:25]: Fixed in [this commit](https://github.com/transferwise/neptune-web/pull/1053/commits/b7c60824ac94e2bdfb7592d4c787b0007f75607d).

**Review by dozortsev — COMMENTED** [review]
noticed that icon in UploadInput looks wrong

Screen Shot 2021-09-10 at 6 08 25 PM

could you please also delete `useDirection` and `isRTL` usages in `UploadButton.tsx` and `UploadItem.tsx`

**Pull request merged**